source/slang/slang-emit-c-like.cpp

```diff
@@ -2169,13 +2224,41 @@ void CLikeSourceEmitter::defaultEmitInstExpr(IRInst* inst, const EmitOpInfo& inO
         }
         else
         {
-            auto prec = getInfo(EmitOp::Postfix);
-            needClose = maybeEmitParens(outerPrec, prec);
+            if (inst->op == kIROp_getElementPtr && supportPtrTypes())
+            {
+                auto ptrType = as<IRPtrTypeBase>(inst->getOperand(0)->getFullType());
+                if (ptrType && ptrType->getValueType() && as<IRArrayTypeBase>(ptrType->getValueType()))
+                {
+                    const auto info = getInfo(EmitOp::Postfix);
+                    needClose = maybeEmitParens(outerPrec, info);
+                    emitOperand(inst->getOperand(0), leftSide(outerPrec, info));
+                    m_writer->emit("->elementPtr(");
+                    emitOperand(inst->getOperand(1), getInfo(EmitOp::General));
+                    m_writer->emit(")");
+                }
+                else
+                {
+                    const auto info = getInfo(EmitOp::Add);
+
+                    needClose = maybeEmitParens(outerPrec, info);
+                    emitOperand(inst->getOperand(0), leftSide(outerPrec, info));
+                    m_writer->emit(" ");
+                    m_writer->emit(info.op);
+                    m_writer->emit(" ");
+                    emitOperand(inst->getOperand(1), rightSide(outerPrec, info));
+                }
```

Comment:
This seems like it can just emit as `&operand0[operand1]` for both cases, rather than needing to special-case either side.

Reply:
Done.